### DISCUSSION TITLE
ScrollBar AllowAutoHide on touch devices should hide scrollbar completely?

### DISCUSSION THREAD

**Discussion by msneijders** [AvaloniaUI/Avalonia#14375] [Ideas]
The `ScrollBar.AllowAutoHide` hides the scrollbar (or collapsing it to a small tracker) on desktop. On Android you only get the small tracker; but `AllowAutoHide` does nothing more. Should it not completely hide the tracker on mobile devices?

**Comment by timunie** [comment]
I think this is a matter of taste. Feel free to add a style for this in your own App. You can probably use platform dependent setters.

**Comment by maxkatz6** [comment]
In some native iOS apps you can still hold your finger on scroll bar and move it. Like, Alarms app. I can’t do it in any app though for some reason.

**Comment by msneijders** [comment]
I have it working in my own apps; although it currently needs reflection because of private setter `ScrolBar.IsExpanded` and private method `ScrollBar.CollapseAfterDelay`; I don't like having to use reflection.

But if the general taste will be to optionally hide it completely, I could justify using time for making a PR suggesting how to do it.

**Reply by timunie** [reply]
```xml
<Style Selector="ScrollBar[IsExpanded=False] /template/ Thumb > Border">
    <Setter Property="IsVisible" Value="{OnPlatform Windows=False , Default=True, x:TypeArguments=x:Boolean}" />
</Style>
```

That hides it for me while not using reflection or breaking the ScrollViewer from opening on pointer over

**Reply by msneijders** [reply]
I was not clear why I use reflection: I set the IsEpanded to True when the scroll position changes; so when you scroll the Thumb is visible, and after scrolling it auto hides again. Currently the ScrollBar sets IsExpanded I believe only on PointerEnter.

**Reply by timunie** [reply]
Got your point. So either IsExpanded or Exaond / Collapse should be public in order to achieve it.

**Reply by msneijders** [reply]
if `ScrollBar.IsExpanded` and `ScrollBar.Expand()`/`ExpandAfterDelay()` and `ScrollBar.Collapse()`/`CollapseAfterDelay()` could become public or protected, I could make my own expanding/collapsing logic and reuse all of ScrollBar.

I probably would add a third state in my custom Scrollbar; currently it has 'Expanded' and 'Collapsed' (IsExpanded=False) and a new state 'Hidden'. So I could make a scrollbar with a big tracker to drag the scrollbar quickly using touch, and a small tracker just for showing scroll position, and also optionally completely auto hide it so it doesn't interfere with other content inside the scrollviewer.
